src/plugins/intel_cpu/src/nodes/rnn.cpp

```diff
@@ -505,6 +506,9 @@ void RNN::configurePortDataTypes() {
     if (one_of(memory::data_type::bf16, inDataTypes[xIdx], inDataTypes[hIdx]))
         inDataTypes[xIdx] = outDataTypes[yIdx] = outDataTypes[hoIdx] = inDataTypes[hIdx] = memory::data_type::bf16; // required by oneDNN.
 
+    if (one_of(memory::data_type::f16, inDataTypes[xIdx], inDataTypes[hIdx]))
+        inDataTypes[xIdx] = outDataTypes[yIdx] = outDataTypes[hoIdx] = inDataTypes[hIdx] = memory::data_type::f32;
```

Comment:
Is it oneDNN requirement?

Reply:
@alvoron it's no oneDNN requirement, onednn can't work with f16 and we move to f32 calculation

Reply:
@EgorDuplensky fixed as temporary solution

Reply:
oneDNN [does not have f16 support|https://oneapi-src.github.io/oneDNN/dev_guide_rnn.html#implementation-limitations] on Intel CPUs due to lack of fast arithmetic. Supported data types are f32, bf16, and int8.